conans/model/build_info.py

```diff
@@ -136,11 +136,17 @@ def __init__(self):
         self._framework_paths = None
         self.version = None  # Version of the conan package
         self.description = None  # Description of the conan package
-        # When package is editable, filter_empty=False, so empty dirs are maintained
-        self.filter_empty = True
+        # By default we don't filter empty, because we want to filter ONLY when the final cppinfo
+        # from package is full of files but before that we will play with the build and source layout
+        # and cppinfo and it mess a lot
+        self.filter_empty = False
```

Comment:
Important: These forced me to change some tests where it assumed the filter empty is always active. 
NOTE: The filter_empty should not belong to the cpp_info, when the final generator get the information of the dep tree it should clear the empty if needed, but not sooner.